examples/kepler-integration/src/components/hubble-export.js

```diff
@@ -0,0 +1,97 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, {Component} from 'react';
+
+// Modal aesthetics
+import {ThemeProvider, withTheme} from 'styled-components';
+import RenderSettingsModal from './render-settings-modal';
+import Button from 'kepler.gl';
+
+// Redux stores/actions
+import {connect as keplerGlConnect} from 'kepler.gl';
+import {connect} from 'react-redux';
+import toggleHubbleExportModal from 'kepler.gl'; // TODO make custom action
+
+// import {createStore, combineReducers, applyMiddleware, compose} from 'redux';
+// import keplerGlReducer from 'kepler.gl/reducers';
+// import {enhanceReduxMiddleware} from 'kepler.gl/middleware';
+
+// const initialState = {};
+// const reducers = combineReducers({
+//   // <-- mount kepler.gl reducer in your app
+//   keplerGl: keplerGlReducer,
+
+//   // Your other reducers here
+// //   app: appReducer
+// });
+
+// // using createStore
+// createStore(
+//   reducers,
+//   initialState,
+//   applyMiddleware(
+//     enhanceReduxMiddleware([
+//       /* Add other middlewares here */
+//     ])
+//   )
+// );
+
+function mapStateToProps(state = {}, props) {
+    return { // TODO unsure if other redux stores are needed atm
+      ...props,
+      visState: state.visState,
+      mapStyle: state.mapStyle,
+      mapState: state.mapState,
+      uiState: state.uiState,
+      providerState: state.providerState,
+    };
+}
+
+// noResultDispatch returns nothing in this case. Undefined if console.log
+// because we're using Kepler's connect (wrapper of Redux connect) it has 3 arguments (2 are dispatches)
+// the code can be found from ../connect/keplergl-connect
+const mapDispatchToProps = () => (noResultDispatch, ownProps, dispatch) => {	
+    return {
+        toggleHubbleExportModal: (isOpen) => dispatch(toggleHubbleExportModal(isOpen)), // NOTE gives dispatch error
+    }
+}	
+
+export class HubbleExport extends Component {
+    handleClose() {this.props.toggleHubbleExportModal(false)} // X button in RenderSettingsModal was clicked
+    handleExport() {this.props.toggleHubbleExportModal(true)} // Export button in Kepler UI was clicked
+
+    render() {
+        return (
+            <div>
+                <RenderSettingsModal isOpen={this.props.uiState.hubbleExportModalOpen} handleClose={this.handleClose.bind(this)} mapData={this.props}/>
+                {/* <ThemeProvider theme={RenderSettingsModal}></ThemeProvider> */}
+                <Button onClick={() => this.handleExport()}>Export</Button> {/* anonymous function to bind state onclick  */}
+            </div>
+        )
+    }
+};
+
+console.log("mapStateToProps", mapStateToProps)
+console.log("mapDispatchToProps", mapDispatchToProps)
+
+// keplerGlConnect is a wrapper of Redux's standard connect w/ access to Kepler's Redux store
+export default connect(mapStateToProps, mapDispatchToProps)(withTheme(HubbleExport));
```

Comment:
You're not using `keplerGlConnect` so wouldn't `mapDispatchToProps` take two arguments?

Reply:
Switched up the connect, mapStateToProps, and mapDispatchToProps. Still not seeing any useful props that are needed for, say, the mapData or uiState
![image](https://user-images.githubusercontent.com/33266041/92820367-bbc68580-f37e-11ea-9c69-971ab0550b6e.png)

```
const mapStateToProps = (state) => state
const mapDispatchToProps = () => (dispatch, props) => {	
    return {
        toggleHubbleExportModal: (isOpen) => dispatch(toggleHubbleExportModal(isOpen)), // NOTE gives dispatch error
    }
}	
```

Reply:
Take a look inside "demo.keplerGl"

Reply:
![image](https://user-images.githubusercontent.com/33266041/92824254-0b0eb500-f383-11ea-8c2a-b8aa454f40b8.png)

Reply:
Gotcha - that's where they should end up once its set up correctly. Did you follow all the steps from the docs? I could look at that code.

Reply:
Please do. I just pushed my latest commit